modules/post/linux/gather/checkvm.rb

```diff
@@ -145,6 +168,48 @@ def run
       end
     end
 
+    # Check BIOS Name
+    if !vm
+      bios_vendor = read_file('/sys/devices/virtual/dmi/id/bios_vendor')
+      if bios_vendor
+        case bios_vendor.gsub("\n", ' ')
+        when /^xen/i
+          vm = 'Xen'
+        end
+      end
+    end
+
+    # Check cpuinfo
+    if !vm
+      cpuinfo = read_file('/proc/cpuinfo')
+      if cpuinfo
+        case cpuinfo.gsub("\n", ' ')
+        when /qemu virtual cpu|emulated by qemu|KVM processor/i
+          vm = 'Qemu/KVM'
+        end
+      end
+    end
+
+    # Check Xen devices
+    if !vm
+      xen_capabilities = read_file('/proc/xen/capabilities')
```

Comment:
Am I right in saying that `/proc/xen/capabilities` would only exist on the host dom0 instance, but not the domU guest VM?

If so - does that mean this logic only detects that the host machine has xen running, rather than detecting that it's being virtualized by Xen - which contradicts the module description:

> 'Description' => %q{
>          This module attempts to determine whether the system is running
>          inside of a virtual environment and if so, which one.

For me it looked like a better detection was checking `cat /sys/hypervisor/uuid` from https://stackoverflow.com/questions/3490995/is-there-an-os-command-i-can-run-to-determine-if-running-inside-a-xen-based-virt

For context, my test set up with `xl list` is:

<img width="361" alt="image" src="https://github.com/rapid7/metasploit-framework/assets/60357436/6d6a6cda-1452-410b-8fe9-3d3f0438f9ed">

Host machine / dom0 (i.e. ubuntu running xen):

<img width="169" alt="image" src="https://github.com/rapid7/metasploit-framework/assets/60357436/5a6f0991-2cf7-4d3f-a05f-c161521019f3">

Guest:

<img width="251" alt="image" src="https://github.com/rapid7/metasploit-framework/assets/60357436/b1706f81-fe09-45a3-92e7-3171f0c84cfb">